language/tools/move-cli/src/lib.rs

```diff
@@ -442,6 +442,66 @@ impl RemoteCache for OnDiskStateView {
     }
 }
 
+/// Holds a closure of modules and provides operations against the closure (e.g., finding all
+/// dependencies of a module).
+pub struct CodeCache {
```

Comment:
- This type + its utilities look pretty useful. There are a lot of `BTreeMap<ModuleId, CompiledModule>`'s in the codebase--it might make sense to put this in `move_core_types` and use this everywhere.
- Rust nit: You can do `pub struct CodeCache(BTreeMap<ModuleId, CompiledModule>)` for single-field structs.